File: Examples/cosmology/sigma.cpp
// ===============================================
// Example code: how to estimate the mass variance
// ===============================================

#include "LCDM.h"
#include "Sigma.h"

using namespace std;

int main () {
    
  try {

    // --------------------------------------------------------
    // ---------------- set the cosmological model ------------
    // --------------------------------------------------------
     
    auto cosmology = make_shared<cbl::cosmology::LCDM>("Planck18");
    

    // --------------------------------------------------------
    // ---------------- estimate the mass variance ------------
    // --------------------------------------------------------

    cbl::cosmology::Sigma SG(cosmology);
    
    const vector<double> mass = {1.e15};
    const double redshift = 1.;
    
    const double sigma2M = SG.sigma2M(mass, "EisensteinHu", redshift)[0];
    
    cout << "The mass variance at M=1e15 Msun and z=1 is: " << sigma2M << endl;

  }

  catch(cbl::glob::Exception &exc) { cerr << exc.what() << endl; exit(1); }

  return 0;
}

File: Examples/cosmology/sigma.py
## ===============================================
## Example code: how to estimate the mass variance
## ===============================================

# to ensure compatibility in Python versions 2.x and 3.x
from __future__ import print_function

# import the CosmoBolognaLib module
import CosmoBolognaLib as cbl

# set the cosmological model, using default parameters
cosmology = cbl.LCDM("Planck18")

# create the object used for the computation of the mass variance
SG = cbl.Sigma(cosmology)

print("The mass variance at M=1e15 Msun and z=1 is:", '%.2f' % SG.sigma2M([1.e15], "EisensteinHu", 1.)[0])
